pydantic/main.py

```diff
@@ -107,58 +106,81 @@ class BaseModel(metaclass=_model_construction.ModelMetaclass):
         __pydantic_private__: Instance attribute with the values of private attributes set on the model instance.
     """
 
-    if TYPE_CHECKING:
-        # Here we provide annotations for the attributes of BaseModel.
-        # Many of these are populated by the metaclass, which is why this section is in a `TYPE_CHECKING` block.
-        # However, for the sake of easy review, we have included type annotations of all class and instance attributes
-        # of `BaseModel` here:
+    # Class attributes:
+    # `model_fields` and `__pydantic_decorators__` must be set for
+    # `GenerateSchema.model_schema` to work for a plain `BaseModel` annotation.
 
-        # Class attributes
-        model_config: ClassVar[ConfigDict]
-        """
-        Configuration for the model, should be a dictionary conforming to [`ConfigDict`][pydantic.config.ConfigDict].
-        """
+    model_config: ClassVar[ConfigDict] = ConfigDict()
+    """
+    Configuration for the model, should be a dictionary conforming to [`ConfigDict`][pydantic.config.ConfigDict].
+    """
 
-        model_fields: ClassVar[dict[str, FieldInfo]]
-        """
-        Metadata about the fields defined on the model,
-        mapping of field names to [`FieldInfo`][pydantic.fields.FieldInfo].
+    # Because `dict` is in local namespace of `BaseModel`, we should use `Dict` for annotations.
```

Comment:
```suggestion
    # Because `dict` is in the local namespace of the `BaseModel` class, we use `Dict` for annotations.
    # TODO v3 fallback to `dict` when the deprecated `dict` method gets removed.
```

nitpick.